src/LockstakeClipper.sol

```diff
@@ -405,24 +418,31 @@ contract LockstakeClipper {
             vat.move(msg.sender, vow, owe);
 
             // Removes Dai out for liquidation from accumulator
-            dog_.digs(ilk, lot == 0 ? tab + owe : owe);
+            dog_.digs(ilk, sale.lot == 0 ? sale.tab + owe : owe);
         }
 
-        if (lot == 0) {
-            uint256 tot = sales[id].tot;
-            engine.onRemove(usr, tot, 0);
+        if (sale.lot == 0) {
+            engine.onRemove(sale.usr, sales[id].tot, 0);
+            uint256 due = sales[id].due;
+            if (due > owe && cuttee != address(0)) {
+                CutteeLike(cuttee).cut(due - owe);
+            }
+            Due -= due;
             _remove(id);
-        } else if (tab == 0) {
-            uint256 tot = sales[id].tot;
-            vat.slip(ilk, address(this), -int256(lot));
-            engine.onRemove(usr, tot - lot, lot);
+        } else if (sale.tab == 0) {
+            vat.slip(ilk, address(this), -int256(sale.lot));
+            engine.onRemove(sale.usr, sales[id].tot - sale.lot, sale.lot);
+            Due -= sales[id].due;
             _remove(id);
         } else {
-            sales[id].tab = tab;
-            sales[id].lot = lot;
+            sales[id].tab = sale.tab;
+            sales[id].lot = sale.lot;
+            uint256 sub = min(sales[id].due, owe);
+            sales[id].due -= sub;
```

Comment:
Could also be `unchecked`. Technically all subtractions from `Due` should be safe as well, since it's the sum of the `due` values.